### PR TITLE
Feature (auth/middleware): handle noSession and add action payload for auth

### PR DESCRIPTION
**Pull request by cerebrl**
Summary:

To handle the parameter of `noSession` in auth trees, we need to detect the completion, successful or not, without the return of an `idToken`. We also need to provide an action payload for app devs to distinguish between auth types and trees.

### CODE REVIEW

File: tests/e2e/app/basic-login/autoscript.js [hunk 1]
```diff
@@ -15,6 +15,23 @@
 
   console.log('Configure the SDK');
   forgerock.Config.set({
+    middleware: [
+      (req, action, next) => {
+        switch (action.type) {
+          case 'START_AUTHENTICATE':
+            if (action.payload.type === 'service' && typeof action.payload.tree === 'string') {
```
Comment (adam-cyclones) [inline tests/e2e/app/basic-login/autoscript.js:22]: If this is

File: tests/e2e/app/basic-login/autoscript.js [hunk 2]
```diff
@@ -15,6 +15,23 @@
...
+              console.log('Starting authentication with service');
+            }
+            break;
+          case 'AUTHENTICATE':
+            if (action.payload.type === 'service' && typeof action.payload.tree === 'string') {
```
Comment (adam-cyclones) [inline tests/e2e/app/basic-login/autoscript.js:27]: the same as this, maybe the if should wrap the switch?

File: src/util/middleware.mock.data.ts [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
+import { ActionTypes } from '../config/enums';
 
 type NextFn = () => RequestObj;
 
 export default [
   (req: RequestObj, action: Action, next: NextFn) => {
     switch (action.type) {
-      case 'a':
-      case 'b':
+      case 'a' as ActionTypes:
```
[the remark this answers is not in the document]
Comment (cerebrl) [reply src/util/middleware.mock.data.ts:9]: Good point, there is a lot of repetition here.

File: tests/e2e/app/basic-login/autoscript.js (hunk 2 above)
Comment (cerebrl) [reply tests/e2e/app/basic-login/autoscript.js:27]: 👍  good point. I get a little lazy in the test files. I'll fix it.

**Pull request merged**